tests/test_GravityCov.pf

```diff
@@ -0,0 +1,125 @@
+
+        r_itrf = (/-3485.799126707284d0,  -5898.652976745232d0, 835.9701786284777d0/)    
+        v_itrf = (/-1.3525457950562447d0, -0.2804534841971075d0, -7.4721873681232385d0/) 
+        time_mjd = 	58324.d0   
+        ! Call the subroutine to be tested
+        call gravity_model%getGravityAcceleration(reduction_handler, r_itrf, v_itrf, time_mjd, accel)
+
+        cov = gravity_model%getGravityCovariance(r_itrf, v_itrf)
+
+        ! Set up expected outputs
+        expected_cov = reshape((/0.39374967173447407d-9, 0.31947373468752591d-8, -0.14864895574315657d-8, &
+        0.31947373468752595d-8, 0.39051986351008705d-8, -0.25245746509934525d-8, &
+        -0.14864895574315659d-8, -0.25245746509934525d-8, -0.42989483068353419d-8/), (/3, 3/))
+
+        ! Set up tolerances
+        tolerance = 1e-10_dp
+
+        ! Check if outputs match expected values
+        @assertRelativelyEqual(cov, expected_cov, tolerance)
+
+        ! Test 2
+        call gravity_model%initGravityPotential('../../work/data')
+
+        r_itrf = (/-3485.799126707284, -5898.652976745232, 835.9701786284777/)    
+        v_itrf = (/-1.3525457950562447, -0.2804534841971075, -7.4721873681232385/) 
+        time_mjd = 	58600.d0   
+        ! Call the subroutine to be tested
+        call gravity_model%getGravityAcceleration(reduction_handler, r_itrf, v_itrf, time_mjd, accel)
+
+        cov = gravity_model%getGravityCovariance(r_itrf, v_itrf)
+
+        ! Set up expected outputs
+        expected_cov = reshape((/0.39374974311657738d-9, 0.31947377200775617d-8, -0.14864897215596233d-8, &
+        0.31947377200775617d-8, 0.39051990534417329d-8, -0.25245749095440029d-8, &
+        -0.14864897215596233d-8, -0.25245749095440029d-8, -0.42989487965583128d-8/), (/3, 3/))
+
+        ! Set up tolerances
+        tolerance = 1e-10_dp
+
+        ! Check if outputs match expected values
+        @assertRelativelyEqual(cov, expected_cov, tolerance)    
+
+        
+        ! Test 3 :
+        call gravity_model%initGravityPotential('../../work/data')
+
+        r_itrf = (/31986.08889084533, 3886.249797563777, -17606.478382491812/)    
+        v_itrf = (/-0.1326762314352175, 2.0844489979862724, -0.13643320490635982/) 
+        time_mjd = 	58324.d0   
+        ! Call the subroutine to be tested
+        call gravity_model%getGravityAcceleration(reduction_handler, r_itrf, v_itrf, time_mjd, accel)
+
+        cov = gravity_model%getGravityCovariance(r_itrf, v_itrf)
+
+        ! Set up expected outputs
+        expected_cov = reshape((/-0.84889040107535929d-12, -0.10613410850253561d-12, -0.11518613812098011d-11, &
+        -0.10613410850253560d-12, 0.40163065395133233d-13, -0.14287823362922449d-12, &
+        -0.11518613812098011d-11, -0.14287823362922446d-12, 0.80872733568022548d-12/), (/3, 3/))
+
+        ! Set up tolerances
+        tolerance = 1e-10_dp
+
+        ! Check if outputs match expected values
+        @assertRelativelyEqual(cov, expected_cov, tolerance) 
```

Comment:
@ArkenCh I think the assertions here are also switched. It should be expected value first then the actual value. Otherwise results for failing tests will be written in a confusing way. (Same probably applies to all other assertions)